src/core/components/dag.js

```diff
@@ -9,6 +9,21 @@ const flattenDeep = require('lodash.flattendeep')
 module.exports = function dag (self) {
   return {
     put: promisify((dagNode, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+      } else if (options.cid && options.format && options.hashAlg) {
```

Comment:
If we change this to `if (options.cid && (options.format || options.hashAlg))` we'll catch cid+format, cid+hashAlg AND cid+format+hashAlg. At the moment if I provide one of `hashAlg` or `format` _as well as_ `cid` it'll fall through to the next check which is for a different restriction.

Reply:
Oh yea, right. Much better. Let me update really quick.